### PR TITLE
Flutter Consent Docs

### PR DESCRIPTION
**Pull request by fabriguespe**
Updated consent docs based on latest flutter [PR](https://github.com/xmtp/xmtp-flutter/pull/94)

- [Preview](https://junk-range-possible-git-flutterconsent-xmtp-labs.vercel.app/docs/build/user-consent)

### CODE REVIEW

File: docs/build/user-consent.md [hunk 1]
```diff
@@ -158,7 +158,12 @@ try await client.contacts.deny(addresses: [spamConvo.peerAddress, unwantedConvo.
 </TabItem>
 <TabItem value="dart" label="Dart"  attributes={{className: "dart_tab"}}>
 
-The user consent feature for Dart hasn't been implemented yet
+```dart
+var auth = AuthManager(client.address, api);
+
+await contacts.deny(auth.keys, wantedConvo.address);
+await contacts.allow(auth.keys, spamConvo.address);
+```
```
Comment (dmccartney) [inline docs/build/user-consent.md:166]: I think instead we should suggest the high-level recipe:
```dart
await client.allowContact(wantedConvo.address);
await client.denyContact(spamConvo.address);
```

File: docs/build/user-consent.md [hunk 2]
```diff
@@ -219,7 +224,12 @@ try await client.contacts.refreshConsentList()
 </TabItem>
 <TabItem value="dart" label="Dart"  attributes={{className: "dart_tab"}}>
 
-The user consent feature for Dart hasn't been implemented yet
+```dart
+var auth = AuthManager(client.address, api);
+var contacts = ContactManager(api, auth);
+
+await contacts.refreshConsents(auth.keys);
+```
```
Comment (dmccartney) [inline docs/build/user-consent.md:232]: high-level is probably better here:
```dart
await client.refreshContactConsentPreferences();
```

File: docs/build/user-consent.md [hunk 3]
```diff
@@ -348,7 +358,13 @@ await client.contacts.isDenied(spamConvo.peerAddress)
...
 <TabItem value="dart" label="Dart"  attributes={{className: "dart_tab"}}>
 
-The user consent feature for Dart hasn't been implemented yet
+```dart
+var auth = AuthManager(client.address, api);
+var contacts = ContactManager(api, auth);
+
+var state = contacts.checkConsent(wantedConvo.address)
+//state can be ContactConsent.deny, ContactConsent.unkown or ContactConsent.allow
+```
```
Comment (dmccartney) [inline docs/build/user-consent.md:367]: consider this high-level call instead:
```dart
var state = client.checkContactConsent(wantedConvo.address);
```

File: docs/build/user-consent.md (hunk 1 above)
Comment (fabriguespe) [reply docs/build/user-consent.md:166]: Ah yes! Found the higher-level testing file. Updated snippets

File: docs/build/user-consent.md [hunk 4]
```diff
@@ -525,3 +535,7 @@ Update a consent preference in the consent list on the network in the following
 The following diagram illustrates the detailed logic for how user consent preferences are set in an app and in the consent list on the XMTP network.
 
 <img src={consentlogic} style={{width:"90%"}}/>
+
+```
+
+```
```
Comment (nplasterer) [inline docs/build/user-consent.md:541]: Is something supposed to be added here or was this an accident?

**Review by nplasterer — APPROVED** [review]

**Pull request merged**